src/IoC/InstancePool.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace IoC;
+
+public sealed class InstancePool
+{
+    //This is where the instances will be stored.
+    private static Dictionary<string, object>? _implementationStore;
+    //Make the contructor unaccessible.
+    InstancePool()
+    {
+        _implementationStore = new Dictionary<string, object>();
+    }
+
+
+    //This is the only way to access to the methods of the class.
+    private static InstancePool? _instance = null;
+    public static InstancePool Instance
+    {
+        get
+        {
+            if (_instance is null)
+            {
+                _instance = new InstancePool();
+            }
+            return _instance;
+        }
+    }
+
+    //This method register the class.
+    public void Register<TClass>() where TClass : class
+    {
+        if (!_implementationStore.ContainsKey(nameof(TClass)))
+        {
+            //Creates the instances before to store it.
+            TClass instance = (TClass)Activator.CreateInstance(typeof(TClass));
+
+            _implementationStore.Add(nameof(TClass), instance);
+        }
+    }
+
+    //This method retrieve the instance of the class previowsly registered.
+    public TClass GetInstance<TClass>() where TClass : class
+    {
+        if (_implementationStore.TryGetValue(nameof(TClass), out object instance))
```

Comment:
Is there a way to make this less painful to understand?

Seems a lot of code to get an instance of a class from a dictionary.

Reply:
The method `TryGetValue` is a safe way to retrieve a value and helps by avoiding a `try` block.

In addition, two values are returned, a `boolean` and the required value if exist in the dictionary.

Reply:
I feel this is too verbose, I feel adding try and catch is more readable if ask to compare each solution

Reply:
Could you wrap this try get into a function to hide implementation details if you want to use it? That way could be more readable, example:

`return Utils.tryAndGet(myTClass)` then do your real try get within the utility

Reply:
Got it. Will have a change on this by tomorrow too.